components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1822,6 +1823,19 @@ void RewardsServiceImpl::OnRecurringDonationsData(const ledger::PublisherInfoLis
   }
 
   callback(list, 0);
+
+  // Incoming ledger::PublisherInfoList needs to be converted to brave_rewards::ContentSiteList
+  brave_rewards::ContentSiteList site_list;
+  for (auto& info : list) {
+    site_list.push_back(PublisherInfoToContentSite(info));
+  }
+
+  TriggerOnRecurringDonations(site_list);
```

Comment:
This method is the response to `GetRecurringDonations` and so should not trigger an observer methods. Observer methods should only be triggered in response to changes, not in response to queries

Reply:
Fixed, This is now done via an asynchronous extension method